### PR TITLE
chore: Use the same pattern for host and group id

### PR DESCRIPTION
**Pull request by gkarat**
This commit edits the OpenAPI schema for HBI: 1) use the same UUID pattern for host and group IDs, 2) fix incorrect reference to PerReporterStaleness.

# Overview

This is a quick patch and it is not linked to any Jiras.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: swagger/api.spec.yaml [hunk 1]
```diff
@@ -1453,19 +1453,29 @@ components:
...
         A group’s human-readable name.
       type: string
       nullable: true
       example: sre-group
+    HostId:
+      description: >-
+        A durable and reliable platform-wide group identifier. Applications
+        should use this identifier to reference hosts.
+      type: string
+      pattern: ^[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}$|^[0-9a-fA-F]{8}[0-9a-fA-F]{4}[0-9a-fA-F]{4}[0-9a-fA-F]{4}[0-9a-fA-F]{12}$
```
Comment (vkrizan) [inline swagger/api.spec.yaml:1470]: Is the pattern really needed if the format is already set to `uuid`?

Comment (gkarat) [reply swagger/api.spec.yaml:1470]: These two options are not equal. `uuid` type guarantees `[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}`-like strings. However, there are some cases where id is generated without dashes, e.g. `3f01b55457674041b75e41829bcee1dc` - `uuid` won't recognize such strings.

Comment (vkrizan) [reply swagger/api.spec.yaml:1471]: Ah, then it depend whether backend can process such uuid strings. I'm not sure if it is that common (or valid).

The second case pattern can be shortened: `^[0-9a-fA-F]{32}$`

Comment (thearifismail) [reply swagger/api.spec.yaml:1471]: @vkrizan I am not what the issue is here.  The groupId and hostId should be uuid, with or without hyphens.   The shortened pattern will not work when dashes are provided.

**Review by thearifismail — COMMENTED** [review]
LGTM.  Running `make style` produced:
```
((gkarat) ) [aarif@daamaan gkarat]$ make style
pre-commit run --all-files
Reorder python imports...................................................Passed
pyupgrade................................................................Passed
black....................................................................Passed
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check Yaml...............................................................Passed
Debug Statements (Python)................................................Passed
Flake8...................................................................Passed
swagger-validation.......................................................Passed
swagger-validation.......................................................Passed
((gkarat) ) [aarif@daamaan gkarat]$ 
```

**Review by thearifismail — DISMISSED** [review]

**Comment by gkarat** [comment]
Something to do with test reporting:
```
ERROR: Step ?Publish JUnit test result report? failed: No test report files were found. Configuration error?
```

**Comment by fstavela** [comment]
> Something to do with test reporting:
> 
> ```
> ERROR: Step ?Publish JUnit test result report? failed: No test report files were found. Configuration error?
> ```

pre-commit failed:
```
10:55:21 Reorder python imports...................................................Passed
10:55:23 pyupgrade................................................................Passed
10:55:30 black....................................................................Passed
10:55:30 Trim Trailing Whitespace.................................................Passed
10:55:30 Fix End of Files.........................................................Passed
10:55:30 Check Yaml...............................................................Passed
10:55:31 Debug Statements (Python)................................................Passed
10:55:35 Flake8...................................................................Passed
10:55:36 swagger-validation.......................................................Failed
10:55:36 - hook id: swagger-validation
10:55:36 - files were modified by this hook
10:55:36 
10:55:36 Created swagger/openapi.json from swagger/api.spec.yaml
10:55:36 
10:55:36 swagger-validation.......................................................Passed
10:55:36 pre-commit ecountered an issue
```
Please, run the pre-commit locally and push changes.

**Review by thearifismail — DISMISSED** [review]
LGTM.  `make style` succeeds on my system:
```
(gkarat) ) [aarif@daamaan gkarat]$ make style
pre-commit run --all-files
Reorder python imports...................................................Passed
pyupgrade................................................................Passed
black....................................................................Passed
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check Yaml...............................................................Passed
Debug Statements (Python)................................................Passed
Flake8...................................................................Passed
swagger-validation.......................................................Passed
swagger-validation.......................................................Passed
((gkarat) ) [aarif@daamaan gkarat]$ 
```

**Comment by kruai** [comment]
@thearifismail Are you on the branch `openapi-schema-reuse-pattern`? The `swagger-validation` step fails for me, and you can see that it fails in the PR build in the Jenkins console as well

**Comment by thearifismail** [comment]
@kruai  Just pushed `openapi.json` that @gkarat  could not generate on his system due to some problem with his setup.  I can merge the branch after the pipeline completes successfully.

**Comment by gkarat** [comment]
@thearifismail, thank you! This looks good and the CI passes. Any objections to merging this PR?

**Review by thearifismail — APPROVED** [review]
LGTM.  Resolved merge conflict caused by PR #1259

**Pull request merged**